File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*; 

class Solution
{
    public int solution(String s)
    {
        ArrayDeque<Character> stack = new ArrayDeque<>();
        
        for(char c : s.toCharArray()) {
            if(stack.isEmpty() || stack.peek() != c) {
                stack.push(c);
            } else {
                stack.pop();
            }
        }
        
        return stack.isEmpty()? 1 : 0;
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(string s)
{
    stack<char> st;
    
    for(char c : s) {
        if(st.empty()) {
            st.push(c);
        } else if (c == st.top()) {
            st.pop();
        } else{
            st.push(c);
        }
    }

    return st.empty();
}